Non lwtey

domsj opened the pull request:
This PR
- removes unneeded lwt-ness
- turn aSSert and aSSert_exists into queries
- fix a concurrency bug in confirm

domsj committed:
make store and backend query functions non lwt-y
turn aSSert and aSSert_exists into queries iso updates
fix concurrency bug in confirm

domsj committed:
remove a couple of warnings

NicolasT commented on src/client/client_protocol.ml:
```diff
@@ -145,46 +145,48 @@ let one_command stop (ic,oc,id) (backend:Backend.backend) =
+    let wrap_exception f =
+      Lwt.catch
+        (fun () ->
+         try f ()
```
I think this try-with is done by Lwt already?

domsj replied:
Hmm, apparently you're right

NicolasT commented on src/client/client_protocol.ml:
```diff
@@ -492,123 +475,112 @@ let one_command stop (ic,oc,id) (backend:Backend.backend) =
+      wrap_exception
         (fun() ->
            Logger.debug_f_ "connection=%s GET_KEY_COUNT" id >>= fun () ->
            backend # get_key_count () >>= fun kc ->
```
I'm somewhat surprised `get` etc are non-Lwt, but `get_key_count` is.

domsj replied:
hmm, somehow 'missed' this one

NicolasT commented on src/client/client_protocol.ml:
```diff
@@ -492,123 +475,112 @@ let one_command stop (ic,oc,id) (backend:Backend.backend) =
+      wrap_exception
         (fun () ->
            Logger.debug_f_ "connection=%s CONFIRM: key=%S" id key >>= fun () ->
            backend # confirm key value >>= fun () ->
```
I'm somewhat surprised `get` etc are non-Lwt, but `confirm` is.

domsj replied:
confirm might do an update, so this should remain Lwt.t

NicolasT commented on src/node/dump_store.ml:
```diff
@@ -21,10 +21,10 @@ open Interval
   Lwt.catch
     (fun () ->
-       f () >>= fun r ->
+       Lwt.wrap f >>= fun r ->
```
Is this `wrap` instead of simply calling?

domsj replied:
I could write `let r = f () in` instead. I wrongly assumed the wrap was needed here, but Lwt.catch has an implicit wrap already...

domsj committed:
comments from 'Non lwtey' PR

domsj commented:
Updated, but I'll run CI again to be sure.

domsj committed:
make some more backend methods non lwt-y
(expect_progress_possible, check, get_routing, get_fringe,
get_cluster_cfgs, get_current_state)

domsj commented:
CI agrees with my latest changes here. Travis failure seems to be camltc related.
